test/node-http-test.js

```diff
@@ -0,0 +1,219 @@
+              week: 0
+            }
+        });
+      });
+      it('It should start an auction on the first day', async () => {
+        await mineBlocks(1);
+        const nameInfo = await nclient.get(`/info/name/${NAME0}`);
+        assert.deepEqual(nameInfo, {
+          info: null,
+            start: {
+              reserved: false,
+              start: 0,
+              week: 0
+            }
+        });
+        const open = await wclient.execute('sendopen', [NAME0]);
+        assert(open);
+      });
+      it('It should start an auction on the 2nd day', async () => {
+        // Question: This test passes non-deterministically. Why?
+        // Note: Keeping this test as proof that the behavior of grindName
+        // isnt working as one would expect.
+        await mineBlocks(175); // Note: This number seems to pass consistently. \o.o/
+        const nameInfo = await nclient.get(`/info/name/${NAME0}`);
+        assert.deepEqual(nameInfo, {
+          info: null,
+            start: {
+              reserved: false,
+              start: 0,
+              week: 0
+            }
+        });
+        const open = await wclient.execute('sendopen', [NAME0]);
+        assert(open);
+        const nameInfoBefore = await nclient.get(`/info/name/${NAME0}`);
+        assert.equal(nameInfoBefore.info, null);
+        await mineBlocks(1);
+        const nameInfoAfter = await nclient.get(`/info/name/${NAME0}`);
+        assert.equal(nameInfoAfter.info.name, NAME0);
+        assert.equal(nameInfoAfter.info.state, 'OPENING');
+      });
+    });
+
+    describe('For names that are available at height 20', () => {
+      it('It should getNameInfo for an opening name', async () => {
+        await mineBlocks(20);
+        await wclient.execute('sendopen', [NAME1]);
+        await mineBlocks(1);
+        const nameInfo = await nclient.get(`/info/name/${NAME1}`);
+        assert(nameInfo.start.start < 20);
+        assert.equal(nameInfo.start.reserved, false);
+        assert.equal(nameInfo.info.state, 'OPENING');
+      });
+    });
+  });
+
+  describe('getNameByHash', () => {
+    it('It should return null when an auction has not been initiated', async () => {
+      const nameHash = rules.hashName(NAME0);
+      const name = await nclient.get(`/name/hash/${nameHash.toString('hex')}`);
+      assert.equal(name, null);
+    });
+
+    describe('When an auction has been initiated', () => {
+      beforeEach(async () => {
+        await mineBlocks(250);
+        await wclient.execute('sendopen', [NAME0]);
+        await mineBlocks(1);
+      });
+      it('It should return the name', async () => {
+        const nameHash = rules.hashName(NAME0);
+        const { name } = await nclient.get(`/name/hash/${nameHash.toString('hex')}`);
+        assert.equal(name, NAME0);
+      });
+    });
+  });
+
+  describe('getNameResource', () => {
+    const zonefile = { compat: false, version: 0, ttl: 172800, ns: ['ns1.example.com.@1.2.3.4'] };
```

Comment:
I'm not sure if its a good idea to call this data `zonefile`, its decoded namestate data. Maybe something like `records`